File: src/doerly-fe/src/app/modules/order/models/responses/get-execution-proposal-response.ts
import { EExecutionProposalStatus } from "../../domain/enums/execution-proposal-status";
import { ProfileInfoModel } from "./profile-info-model";

export interface GetExecutionProposalResponse { 
    id: number;
    orderId: number;
    comment: string | null;
    senderId: number;
    sender: ProfileInfoModel | null;
    receiverId: number;
    receiver: ProfileInfoModel | null;
    status: EExecutionProposalStatus;
}

File: src/main-service/Doerly.Module.Order/Doerly.Module.Order.DataTransferObjects/Responses/ExecutionProposal/GetExecutionProposalResponse.cs
﻿using Doerly.Module.Order.Enums;
using Doerly.Module.Profile.DataTransferObjects.Profile;

namespace Doerly.Module.Order.DataTransferObjects.Responses;
public class GetExecutionProposalResponse
{
    public int Id { get; set; }

    public int OrderId { get; set; }

    public string Comment { get; set; }

    public int SenderId { get; set; }

    public ProfileInfo Sender { get; set; }

    public int ReceiverId { get; set; }

    public ProfileInfo Receiver { get; set; }

    public EExecutionProposalStatus Status { get; set; }

    public DateTime DateCreated { get; set; }
}
